src/rl78.c

```diff
@@ -131,7 +197,7 @@ int rl78_send_cmd(port_handle_t fd, int cmd, const void *data, int len)
     buf[len + 4] = ETX;
     int ret = serial_write(fd, buf, sizeof buf);
     // Read back echo
-    if (1 == communication_mode)
+    if ((1 == communication_mode) | (5 == communication_mode))
     {
         serial_read(fd, buf, sizeof buf);
     }
```

Comment:
Well, damn.  This isn't in the same function, so `r` is out of scope.

It might make sense to make a function:
```bool is_1wire() 
{ 
    return (1 == communication_mode) || (5 == communication_mode); 
}
```

Note that's a `||` double pipe, not single. Single pipe is a bit-wise OR, double pipe is a bool operator.  We want the bool here.

You don't even have to pass in `communication_mode` because it's a static global.  It's always available.